antelope_contracts/contracts/erc20/src/erc20.cpp

```diff
@@ -8,92 +8,269 @@
 #include <silkworm/core/execution/address.hpp>
 #include <silkworm/core/common/util.hpp>
 
+namespace eosio {
+   namespace internal_use_do_not_use {
+      extern "C" {
+         __attribute__((eosio_wasm_import))
+         uint32_t get_code_hash(uint64_t account, uint32_t struct_version, char* data, uint32_t size);
+      }
+   }
+}
+
 namespace erc20 {
 
+checksum256 get_code_hash(name account) {
+    char buff[64];
+
+    eosio::check(internal_use_do_not_use::get_code_hash(account.value, 0, buff, sizeof(buff)) <= sizeof(buff), "get_code_hash() too big");
+    using start_of_code_hash_return = std::tuple<unsigned_int, uint64_t, checksum256>;
+    const auto& [v, s, code_hash] = unpack<start_of_code_hash_return>(buff, sizeof(buff));
+
+    return code_hash;
+}
+
 template <size_t Size>
 void initialize_data(bytes& output, const unsigned char (&arr)[Size]) {
+    static_assert(Size > 128); // ensure bytecode is compiled
     output.resize(Size);
     std::memcpy(output.data(), arr, Size);
 }
 
+uint64_t erc20::my_next_nonce() { // lookup nonce from the multi_index table of evm contract
+    nextnonces_table_t table(evm_account, evm_account.value);
+    auto itr = table.find(get_self().value);
+    if (itr == table.end()) return 0;
+    else return itr->next_nonce;
+}
+
 void erc20::init() {
     require_auth(get_self());
 
     bytes call_data;
 
     auto reserved_addr = silkworm::make_reserved_address(get_self().value);
     initialize_data(call_data, solidity::erc20::bytecode);
+
     bytes to = {};
-    // Assume account opened in evm_runtime
+    bytes value_zero; 
+    value_zero.resize(32, 0);
+
+    // required account opened in evm_runtime
     call_action call_act(evm_account, {{get_self(), "active"_n}});
-    call_act.send(get_self(), to, 0, call_data, evm_init_gaslimit);
+    call_act.send(get_self(), to, value_zero, call_data, evm_init_gaslimit);
+
+
+    evmc::address impl_addr = silkworm::create_address(reserved_addr, my_next_nonce()); 
```

Comment:
Send inline `assertnonce` action to ensure your retrieved nonce used to calculate the `impl_addr` is correct.

Reply:
fixed